pylzy/lzy/env/environment.py

```diff
@@ -0,0 +1,78 @@
+# pylint: disable=too-many-instance-attributes
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Dict, TypeVar, TYPE_CHECKING
+from typing_extensions import Self, final
+
+from .base import Deconstructible, NotSpecified, EnvironmentField, is_specified
+from .container.base import BaseContainer
+from .container.no_container import NoContainer
+from .provisioning.provisioning import Provisioning
+from .python.auto import AutoPythonEnv
+from .python.base import BasePythonEnv
+
+if TYPE_CHECKING:
+    from .mixin import WithEnvironmentType
+
+
+T = TypeVar('T')
+EnvVarsType = Dict[str, str]
+
+
+@final
+@dataclass
+class LzyEnvironment(Deconstructible):
```

Comment:
let's drop `Deconstructible`

Reply:
No-no-no, Deconstructible is mega useful.
I am using at different places:
1) `__hash__`
2) `.with_fields`
3) `__repr__`
4) `__eq__`